cpu/atxmega/periph/i2c.c

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright (C) 2021 Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_atxmega
+ * @ingroup     cpu_atxmega_periph
+ * @{
+ *
+ * @file
+ * @brief       Low-level I2C driver implementation
+ *
+ * @author      Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * @}
+ */
+#include <errno.h>
+
+#include "cpu.h"
+#include "cpu_pm.h"
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+#include "periph/pm.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+/**
+ * @brief   Device Context allocation
+ */
+static struct {
+    mutex_t locks;
+    mutex_t xfer;
+    uint8_t *buffer;
+    size_t pos;
+    size_t len;
+    uint8_t flags;
+    volatile int8_t status;
```

Comment:
No need for `volatile` here, you are waiting on `xfer` before accessing it.

Reply:
removed!